crates/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -690,7 +690,18 @@ impl<'interner> Monomorphizer<'interner> {
                 let args = vecmap(args, Self::convert_type);
                 let ret = Box::new(Self::convert_type(ret));
                 let env = Box::new(Self::convert_type(env));
-                ast::Type::Function(args, ret, env)
+                match (*env).clone() {
+                    ast::Type::Unit => ast::Type::Function(args, ret, env),
+                    ast::Type::Tuple(elements) => ast::Type::Tuple(vec![
+                        elements[0].clone(),
+                        ast::Type::Function(args, ret, env),
+                    ]),
```

Comment:
Why are we assuming the closure environment only has 1 item? I'd think we'd need all the `elements` kept in their own tuple.

Reply:
ok, just reused `env` directly